modules/Link.js

```diff
@@ -119,7 +118,7 @@ const Link = React.createClass({
       if (activeClassName || (activeStyle != null && !isEmptyObject(activeStyle))) {
         if (router.isActive(location, onlyActiveOnIndex)) {
           if (activeClassName)
-            props.className += props.className === '' ? activeClassName : ` ${activeClassName}`
+            props.className = `${props.className || ''} ${activeClassName}`.trim()
```

Comment:
Agreed. We can do that after merge. For now, this gets the job done and adds a test so any further fixes are easier.